include/config.h

```diff
@@ -0,0 +1,28 @@
+#ifndef CONFIG_H
+#define CONFIG_H
+
+#include "logger.h"
+
+struct config_general {
+	char *configfile;
+};
+
+struct config_logger {
+	enum LOGLEVEL loglevel;
```

Comment:
I wonder, is there a use-case for setting the log level in the config file? The logging level mostly exists for debugging purposes, which means manually starting xdpw.

For instance, Sway doesn't have an option to set it from the config file.

Reply:
My idea was to put every option we have into one central config structure. If we want, we can read it from the configfile, of not, we wont.

Reply:
I don't think that's worthwhile if that's just adding code for no real purpose.

Reply:
I kind of like this because starting xdpw manually can be a pain. This would change verbosity in journald if it is started via dbus activation as a systemd user service.

This would mean people running master could keep log verbosity up all the time without overriding the service file.

Reply:
I'm not sure I understand the use-case here.

You almost never want DEBUG/TRACE levels when running with D-Bus activation, because that'd fill up your hard drive. When running on master, you likely don't have the version you compiled installed, otherwise it would conflict with the one from your package manager. Even if you want DEBUG/TRACE and you have master installed to /usr, then you'd need to edit your config file, restart xdpw (how?), reproduce the bug, then use journalctl to watch/collect the logs, then revert the config file change.

It sounds a lot easier to just run xdpw from the build directory, with the log level flag, redirecting the output to a file.

Reply:
@emersion I think the problem is that when we ask people in issues to give us logs, they often get confused by the process of killing all instances of xdp/xdpw and starting both from the command line in such a way that the right instances are picked up and used. They're not often building from source. I've seen people suggesting overriding the systemd units just to get logs. If you're developing xdpw, running from the command line is a must, but for regular users, this is confusing.

I think your recent patch to add the `-r` flag will help a lot with this and may invalidate my concerns. Maybe, in addition to this, improved documentation would help in the FAQ or README.

Even when I'm testing a PR, I typically hack a PKGBUILD together that points to another fork so that I can rely on d-bus activation for a few days to test out the stability of the changes. It is nice to have logging up a little (not TRACE level, but DEBUG isn't *too* bad). This helps for more pervasive issues that are harder to catch other than by regular use.

The more I write this out, the more I think wanting a config file option for this is the exception and not the rule. Overriding the systemd unit may not be such a bad thing to have to do in the exceptional case.

Reply:
I think adding the -r flag will make this issue invalid, since the default loglevel is error, so the log will hopefully contain the reason of failure and for a more detailed log the installed version of xdpw can be used with --reload -l DEBUG/TRACE.
The only usecase left would be people which want reduce the loglevel to QUIET.